engine/development/deploymentManager/src/main/scala/pl/touk/nussknacker/development/manager/DevelopmentDeploymentManagerProvider.scala

```diff
@@ -0,0 +1,140 @@
+
+  private val MinSleepTime = 5
+  private val MaxSleepTime = 12
+
+  private val memory: TrieMap[ProcessName, ProcessState] = TrieMap[ProcessName, ProcessState]()
+  private val random: ThreadLocalRandom = ThreadLocalRandom.current()
+
+  implicit private class ProcessStateExpandable(processState: ProcessState) {
+    def withStateStatus(stateStatus: StateStatus): ProcessState = {
+      val processStateForStateStatus = processStateDefinitionManager.processState(
+        stateStatus, startTime = Some(System.currentTimeMillis())
+      )
+
+      processStateForStateStatus.copy(
+        deploymentId = processState.deploymentId,
+        version = processState.version
+      )
+    }
+  }
+
+  override def deploy(processVersion: ProcessVersion, deploymentData: DeploymentData, canonicalProcess: CanonicalProcess, savepointPath: Option[String]): Future[Option[ExternalDeploymentId]] = {
+    logger.debug(s"Starting deploying scenario: ${processVersion.processName}..")
+    val duringDeployStateStatus: ProcessState = createStateStatus(SimpleStateStatus.DuringDeploy, processVersion)
+    memory.update(processVersion.processName, duringDeployStateStatus)
+    asyncChangeState(processVersion.processName, SimpleStateStatus.Running)
+    logger.debug(s"Finished deploying scenario: ${processVersion.processName}.")
+    Future.successful(duringDeployStateStatus.deploymentId)
+  }
+
+  override def stop(name: ProcessName, savepointDir: Option[String], user: User): Future[SavepointResult] = {
+    logger.debug(s"Starting stopping scenario: $name..")
+    asyncChangeState(name, SimpleStateStatus.Finished)
+    logger.debug(s"Finished stopping scenario: $name.")
+    Future.successful(SavepointResult(""))
+  }
+
+  override def cancel(name: ProcessName, user: User): Future[Unit] = {
+    logger.debug(s"Starting canceling scenario: $name..")
+    changeState(name, SimpleStateStatus.DuringCancel)
+    asyncChangeState(name, SimpleStateStatus.Canceled)
+    logger.debug(s"Finished canceling scenario: $name.")
+    Future.unit
+  }
+
+  override def test[T](name: ProcessName, canonicalProcess: CanonicalProcess, testData: TestData, variableEncoder: Any => T): Future[TestProcess.TestResults[T]] = ???
+
+  override def findJobStatus(name: ProcessName): Future[Option[ProcessState]] =
+    Future.successful(memory.get(name))
+
+  override def savepoint(name: ProcessName, savepointDir: Option[String]): Future[SavepointResult] =
+    Future.successful(SavepointResult(""))
+
+  override def processStateDefinitionManager: ProcessStateDefinitionManager = SimpleProcessStateDefinitionManager
+
+  override def customActions: List[CustomAction] = List(
+    CustomAction("test-running", List("RUNNING")),
+    CustomAction("test-canceled", List("CANCELED")),
+    CustomAction("test-all", Nil),
+  )
+
+  override def invokeCustomAction(actionRequest: CustomActionRequest, canonicalProcess: CanonicalProcess): Future[Either[CustomActionError, CustomActionResult]] =
```

Comment:
Can't you do `asyncChangeState` in this action invocations? (based on `canonicalProcess.id`)? I thought that it is the main purpose of this class - to be able to see that custom action "do sth". How u will know that something was happen?